kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/components/menus/PublicSelectMenu.kt

```diff
@@ -20,22 +19,19 @@ import dev.kord.common.annotation.KordUnsafe
 import dev.kord.core.behavior.interaction.modal
 import dev.kord.core.behavior.interaction.respondEphemeral
 import dev.kord.core.behavior.interaction.respondPublic
+import dev.kord.core.behavior.interaction.response.PublicMessageInteractionResponseBehavior
 import dev.kord.core.event.interaction.SelectMenuInteractionCreateEvent
-import dev.kord.rest.builder.message.create.InteractionResponseCreateBuilder
-
-public typealias InitialPublicSelectMenuResponseBuilder =
-    (suspend InteractionResponseCreateBuilder.(SelectMenuInteractionCreateEvent) -> Unit)?
 
-/** Class representing a public-only select (dropdown) menu. **/
-public open class PublicSelectMenu<M : ModalForm>(
+/** Class for public select menus. **/
+public abstract class PublicSelectMenu<C, M : ModalForm>(
```

Comment:
Since `C` needs to meet 2 conditions, it is defined in the where clause a few lines below (`where C : SelectMenuContext, C : PublicInteractionContext`)